musit-models/src/main/scala/no/uio/musit/models/Module.scala

```diff
@@ -0,0 +1,25 @@
+package no.uio.musit.models
+
+import play.api.libs.json.{JsNumber, Reads, Writes, __}
+
+sealed trait Module {
+  val id: Int
+}
+
+case object StorageFacility extends Module {
+  val id = 1
+}
+case object LoanProcess extends Module {
+  val id = 2
+}
+
+object Module {
+
+  implicit val reads: Reads[Module]   = __.read[Int].map(fromInt)
+  implicit val writes: Writes[Module] = Writes{m => JsNumber(m.id)}
+
+  def fromInt(id: Int): Module = id match {
+    case StorageFacility.id => StorageFacility
```

Comment:
I am not very convinced about this, at all. All the rows in the db column GROUP_MODULE will either be set by migration script or validated in through the form controllers for the group admin gui. So I really dont see why we should need to wrap an exception that will never happen. And exception if it ever occurs will be non exhaustive match exception given the excact line number and place in the code. This will tell us whats the problem. We dont need NoSuchElementException for that. Its overengineering